heron/spi/src/java/com/twitter/heron/spi/utils/ShellUtils.java

```diff
@@ -111,11 +112,24 @@ public static Process runASyncProcess(
       boolean verbose, String[] command, File workingDirectory) {
     // Log the command for debugging
     LOG.log(Level.FINE, "$> {0}", Arrays.toString(command));
+    // Log the working directory, this will make people fast locate the
+    // directory to find the log files
+    LOG.log(Level.INFO, "working dir: {0}", workingDirectory.getAbsolutePath());
+
+    // the log file can help people to find out what happened between pb.start()
+    // and the async process started
+    String commandFileName = Paths.get(command[0]).getFileName().toString();
+    String uuid = UUID.randomUUID().toString().substring(0, 8);
+    String logFilePath = workingDirectory + "/" + commandFileName + "-" + uuid + "-started.stderr";
+    File logFile = new File(logFilePath);
 
     // For AsyncProcess, we will never inherit IO, since parent process will not
     // be guaranteed alive when children processing trying to flush to
     // parent processes's IO.
     ProcessBuilder pb = getProcessBuilder(false, command, workingDirectory);
+    pb.redirectErrorStream(true);
+    pb.redirectOutput(ProcessBuilder.Redirect.appendTo(logFile));
```

Comment:
Hi @ashvina.
I find a simple way to do this, as long as we capture the output to the yarn's logging system and the log file is no longer necessary.
```
LOG.log(Level.FINE, "\tSTDOUT: {0}", inputstreamToString(process.getInputStream()));
LOG.log(Level.SEVERE, "\tSTDERR: {0}", inputstreamToString(process.getErrorStream()));
```
This works fine too. What do you think? If you agree with me, I'll put a new commit.